modules/gapi/samples/pipeline_modeling_tool/utils.hpp

```diff
@@ -91,6 +93,26 @@ typename duration_t::rep timestamp() {
     return duration_cast<duration_t>(now.time_since_epoch()).count();
 }
 
+#define RETHROW_WITH_MSG_IF_FAILED(expr, msg)      \
+    try {                                          \
+        expr;                                      \
+    } catch (const std::exception& e) {            \
+        std::stringstream ss;                      \
+        ss << msg << "\n caused by: " << e.what(); \
+        throw std::logic_error(ss.str());          \
+    }                                              \
```

Comment:
What is the reason to do this? Maybe @sivanov-work have more to say here? :)

Reply:
Answered above

Reply:
Removed ugly macro